src/twisted/test/test_failure.py

```diff
@@ -24,6 +25,18 @@
 from twisted.trial.unittest import SynchronousTestCase
 
 
+class ComparableException(Exception):
+    """An exception that can be compared by value."""
+
+    def __eq__(self, other: object) -> bool:
+        return (self.__class__ == other.__class__) and (
+            self.args == cast(ComparableException, other).args
+        )
+
+    def __ne__(self, other: object) -> bool:
+        return not self.__eq__(other)
```

Comment:
It looks like this code is not used. Can we removed it ?

Reply:
My worry is someone tweaks the test logic slightly and then it starts acting weirdly because of the lack of `__ne__`. But I just learned that they're optional in Python 3, lacking one it will DTRT, so will do.